### DISCUSSION TITLE
Distribute test cases on multiple mobile devices

### DISCUSSION THREAD

**Discussion by akhil1809** [webdriverio/webdriverio#8528] [General]
### Discussed in https://github.com/webdriverio/webdriverio/discussions/8076

<div type='discussions-op-text'>

<sup>Originally posted by **kitoymonster** March  3, 2022</sup>
Hi guys.

Is it possible to distribute test case execution on multiple android devices? So far, I have already configured the test to run on multiple devices but it's testing the same test cases so my 7 test will generate 14 workers. What I wanted to achieve is to distribute the test to multiple devices so the overall execution time will be less. This same behavior is available in webdriverio on browser based test.

When I set `maxInstances = 2` in conf and I only have 1 device in caps, I am able to see the desired behavior but because I only have one device then the test will fail since both test will be using the same device at the same time.
```
config.capabilities = [
  {
    platformName: 'Android',
    acceptInsecureCerts: true,
    autoGrantPermissions: true,
    platformVersion: '11.0',
    automationName: 'UiAutomator2',
    appPackage: 'com.test.app',
    appActivity: 'com.test.app.MainActivity',
    noReset: false
  }
]
```

```
Execution of 7 workers started at 2022-03-03T06:14:55.827Z
[0-1] RUNNING in com.test.app - /test/regression/profile/Test1.e2e.js
[0-0] RUNNING in com.test.app - /test/regression/profile/Test2.e2e.js
```

But when I set `maxInstances = 2` and set 2 devices, it just test the same test in both devices.
```
let commonCaps = {
  platformName: 'Android',
  autoGrantPermissions: true,
  orientation: 'PORTRAIT',
  automationName: 'UiAutomator2',
  // The path to the app
  app: join(process.cwd(), './apps/app-debug.apk'),
  appPackage: 'com.test.app',
  appActivity: 'com.test.app.MainActivity',
  noReset: false
}

config.capabilities = [
  {
    ...commonCaps,
    udid: '192.168.56.102:5555'
  },{
    ...commonCaps,
    udid: '192.168.56.104:5555'
  },
]

exports.config = config
```

```
Execution of 14 workers started at 2022-03-03T06:06:33.438Z
[0-0] RUNNING in com.test.app - /test/regression/profile/Test1.e2e.js
[0-1] RUNNING in com.test.app - /test/regression/profile/Test1.e2e.js
```
</div>

**Comment by christian-bromann** [comment]
This is not something WebdriverIO manages. You should check out the Appium project and set up a grid with your devices.
